presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -148,23 +148,29 @@ public void run()
         }
     }
 
-    private String promptFor(boolean password)
+    private Optional<String> getPassword()
     {
-        if (password) {
-            checkState(clientOptions.user != null, "Username must be specified along with password");
-            ConsoleReader consoleReader = null;
-            try {
-                consoleReader = new ConsoleReader();
-                consoleReader.setExpandEvents(false);
-                return consoleReader.readLine("Password: ", '*');
-            }
-            catch (IOException e) {
-                System.err.println("Console read error: " + e.getMessage());
-            }
-            finally {
-                if (consoleReader != null) {
-                    consoleReader.shutdown();
-                }
+        checkState(clientOptions.user != null, "Username must be specified along with password");
```

Comment:
I think this requires the user to always be specified, you should change this checkstate if that's the case